StaticResourceSources/BGEResources/js/grid/app.js

```diff
@@ -583,6 +584,31 @@
             }
         }
 
+        function afterBeginEditingHandler(row, col) {
+
+            var colType = hot.getDataType(row, col);
```

Comment:
You got the column type before calling this method, you can pass the column type as a parameter here and avoid repeating the operation if the parameter is not null, empty, undefined or whitespace.